<launch>
  
  <!-- Start the VRPN Client (only needed if recording data)-->
  <!--
  <include file="/opt/ros/kinetic/share/vrpn_client_ros/launch/OF_F330.launch"/>
  -->

  <!-- Start coms to PixHawk via Mavros -->
  <include file="/opt/ros/kinetic/share/mavros/launch/px4.launch"/>

  <!-- Start the camera node -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="320" />
    <param name="image_height" value="240" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="framerate" value="30"/>
  </node>
  
  <!-- Start the optic flow node -->
  <node pkg="small_object" type="OF_controller.py" name="OF_node" output="screen" />

  <!-- Start the flight code (steering controller) -->
  <node pkg="flight_code" type="small_object_flight_code" name="flight_control" output="screen" />


<!-- RECORD TEMPLATE -->

  <!-- Start recording with Rosbag 
  <node pkg="rosbag" type="record" name="rosbag_OF_data"
        args=" -o /tmp/rosbag_OF_data /optic_flow/optic_flow /optic_flow/fourier_coefs " />
  <node pkg="rosbag" type="record" name="rosbag_flight_data"
        args=" -o /tmp/rosbag_flight_data  /mavros/mocap/pose /mavros/rc/in " />
-->
</launch>
